### DISCUSSION TITLE
Accessing a Reference Datum In Eopsin

### DISCUSSION THREAD

**Discussion by while0x1** [OpShin/opshin#41] [Q&A]
Are there any examples demonstrating how to access a datum sent to the contract address in a previous transaction?

**Comment by nielstron** [comment]
Not that I am aware of. The examples usually consider the datum being inside the UTxO being spent. Datums from other inputs need to be resolved via their datum hash and txinfo.data

**Reply by while0x1** [reply]
Trying to establish a way to define some spending constraints in a datum but I also want to make sure that only a certain address or policy key holder can define those constraints
